shark_turbine/kernel/wave/constraints.py

```diff
@@ -176,3 +176,41 @@ def apply(self) -> IndexSequence:
                 "Index is being computed without setting induction variable"
             )
         return IndexSequence(self.induction_var * self.tile_size, 1)
+
+
+@dataclass
+class WaveConstraint(Constraint):
```

Comment:
@harsh-nod is this the lowest level of constraint we expose to users? Since based on this we can determine indexing based of threadIdx.

If the above is true, is that why we called it TKW?

Reply:
Yes this is the lowest level of constraint we expose to users. The reason the lang is called tkw is because the kernels are authored at the wave level, but the reason why we need this constraint is because the author needs to specify how many waves will operate along each dimension (or more specifically what is the tile size that each wave is operating on, which implicitly translates to how many waves operate along each dimension).